packages/core/src/synchronizer.ts

```diff
@@ -157,16 +157,17 @@ export class UnirepSocialSynchronizer extends Synchronizer {
                 },
                 update: {
                     status: 1,
-                    transactionHash: _transactionHash,
+                    transactionHash,
                 },
             })
         } else {
+            const hashedContent: string = decodedData._postContent ?? ''
```

Comment:
We should never get an `undefined` or `null` post content hash. We should also check the hash against the actual content to make sure it matches.